external-crates/move/crates/move-package-alt/src/package/package_impl.rs

```diff
@@ -418,33 +438,34 @@ mod tests {
     }
 
     #[test]
-    fn test_explicit_implicit_deps_with_invalid_names() {
+    /// Test that we can also "add" deps that are not in the default list of the flavor.
+    fn test_explicit_deps_with_more_than_default_names() {
         let env = test_environment();
-        let implicit_deps =
-            ImplicitDepMode::Enabled(Some(vec!["ignore".to_string(), "foo".to_string()]));
+        let implicit_deps = Some(vec!["foo".to_string(), "baz".to_string()]);
 
-        assert!(Package::<TestFlavor>::implicit_deps(&env, implicit_deps).is_err());
+        let deps = Package::<TestFlavor>::system_dependencies(&env, implicit_deps).unwrap();
+        let dep_keys: Vec<_> = deps.keys().cloned().collect();
+
+        assert_eq!(dep_keys.len(), 2);
+        assert!(dep_keys.contains(&new_package_name("foo")));
+        assert!(dep_keys.contains(&new_package_name("baz")));
     }
 
     #[test]
-    /// We disable implicit deps.
-    fn test_no_implicit_deps() {
+    fn test_explicit_system_deps_with_invalid_names() {
         let env = test_environment();
-        let implicit_deps = ImplicitDepMode::Disabled;
+        let implicit_deps = Some(vec!["ignore".to_string(), "foo".to_string()]);
 
-        let deps = Package::<TestFlavor>::implicit_deps(&env, implicit_deps).unwrap();
-
-        assert_eq!(deps.len(), 0);
+        assert!(Package::<TestFlavor>::system_dependencies(&env, implicit_deps).is_err());
```

Comment:
```suggestion
        assert_snapshot!(
            Package::<TestFlavor>::system_dependencies(&env, implicit_deps).unwrap_err(),
            @"The system dependency `ignore` does not exist in the implicit deps list."
        );
```

Reply:
I particularly like the snapshot tests for user-facing error messages since they make it easy to review the error message